src/shim/ShimUtils.ts

```diff
@@ -111,19 +111,19 @@ export function validateDocumentData(data: Uint8Array) {
 }
 
 /**
- * Validate that the provided encrypted document is in the proper form, which should be Base64 encoded bytes of the document content and the AES IV prepended. Therefore
- * validate the length is at least 1 more than the IV length.
+ * Validate that the provided encrypted document is in the proper form, which should be Base64 encoded bytes or a Uint8Array of the document content and the AES IV prepended.
+ * Therefore validate the length is at least 1 more than the IV length.
  */
-export function validateEncryptedDocument(documentData: Base64Bytes) {
-    if (typeof documentData !== "string") {
-        throw new Error(`Invalid encrypted document format provided. Expected a string.`);
-    }
+export function validateEncryptedDocument(documentData: Base64Bytes | Uint8Array) {
     let encryptedByteLength: number;
+    if (typeof documentData !== "string" && documentData.constructor.name !== "Uint8Array") {
```

Comment:
Pretty minor, but I think it's better to do `data instanceof Uint8Array` here instead of checking `constructor.name`.